tests/test_INCA_script_generator.py

```diff
@@ -0,0 +1,56 @@
+import unittest
+import pickle
+import sys
+import pathlib
+
+sys.path.insert(1, "../")
```

Comment:
You did an amazing job of refactoring the code into discrete methods.  I would recommend adding in unit tests for each of the discrete functions so that if something goes wrong, we know exactly which of the refactored functions is the cause.